homeassistant/components/epson/__init__.py

```diff
@@ -60,5 +61,33 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
-        hass.data[DOMAIN].pop(entry.entry_id)
+        projector = hass.data[DOMAIN].pop(entry.entry_id)
+        projector.close()
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug(
+        "Migrating configuration from version %s.%s",
+        config_entry.version,
+        config_entry.minor_version,
+    )
+
+    if config_entry.version > 1 or config_entry.minor_version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if config_entry.version == 1 and config_entry.minor_version == 1:
+        new_data = {**config_entry.data}
+        new_data[CONF_CONNECTION_TYPE] = "http"
```

Comment:
So wait, in the config flow you add the option for users to select `HTTP` or `Serial` and in the migration we set it to `http`? Let's make the config flow also provide lower_case names and use translations to make it look nice for the user.